config/cdc/postgres-entrypoint.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+set -e
+
+WAL2JSON_VERSION=0.0.1
+
+cdc_setup_hba_conf() {
+    # Ensure pg-hba is properly configured
+
+    PG_HBA="$PGDATA/pg_hba.conf"
+    if [ ! -f "$PG_HBA" ]; then
+        echo "DB not initialized. Postgres will take care of pg_hba"
+    fi
+
+    if [ "$(grep -c -E "^host\s+replication" "$PGDATA"/pg_hba.conf)" -ge 0 ]; then
+        echo "Replication config already present in pg_hba"
+    else
+        echo "host replication all all $POSTGRES_HOST_AUTH_METHOD" >> "$PGDATA/pg_hba.conf"
+    fi
+}
+
+install_wal2json() {
+    # Install the pinned version of wal2json if it is not already
+    # present in /wal2json
+
+    ARCH=$(uname -m)
+    FILE_NAME="wal2json-Linux-$ARCH.so"
+
+    if [ ! -f "/wal2json/$WAL2JSON_VERSION/$FILE_NAME" ]; then
+        mkdir -p "/wal2json/$WAL2JSON_VERSION"
+        wget \
+            "https://github.com/getsentry/wal2json/releases/download/$WAL2JSON_VERSION/$FILE_NAME" \
+            -P "/wal2json/$WAL2JSON_VERSION/"
+    fi
+
+    cp "/wal2json/$WAL2JSON_VERSION/$FILE_NAME" /usr/local/lib/postgresql/wal2json.so
+
+    echo "wal2json installed"
+}
+
+echo "Setting up Change Data Capture"
+
+if [ "$1" = 'postgres' ]; then
```

Comment:
Is there a time where we *won't* call this script with `postgress` as the first parameter?

This block makes me think that we always pass the command, thus, we don't ever call this entrypoint *without* entering this block.

```shell
        "command": [
            "postgres",
            "-c",
            "wal_level=logical",
            "-c",
            "max_replication_slots=1",
            "-c",
            "max_wal_senders=1",
        ],
```

Reply:
there are other possible commands (Like here https://github.com/getsentry/sentry/blob/master/src/sentry/conf/server.py#L1600 to check if postgres is ready). I am not sure any makes sense before postgres is running, so I did not think doing the initialization work would be really needed if we are not starting postgres. 

Basically copied this pattern from the default postgres entrypoint.